### DISCUSSION TITLE
Is it possible to trigger the app to run specific JS during tests?

### DISCUSSION THREAD

**Discussion by RhyG** [mobile-dev-inc/Maestro#968] [Q&A] [answered]
Currently trying out Maestro to potentially migrate from a fairly extensive Detox suite that has been a massive pain to implement and maintain. As part of that we run a lot of "setup" when starting an E2E test, depending on the test. For example we do things like:
- Log the user in or out to avoid doing it manually in every test.
- Dispatch some redux actions to set values in the store needed for the test.
- Enable or disable specific user settings depending on the test.

These are done by running JS setup files that update Redux and context for what's needed. We do this by sending a "signal" via notification to the app which tells it what to run, but Maestro right now doesn't support notifications.

I'm wondering if it's possible to do the same with Maestro. I've seen the `runScript` command, but seems like it's not quite the same. Any help would be super appreciated because so far Maestro has been amazing to use and would love to be able to migrate.

**Comment by axelniklasson** [comment, marked as answer]
Hi @RhyG! Yes, `runScript` supports conditional execution where you can pass for example variables into your JS file to do what you want at the right time - more information [here](https://maestro.mobile.dev/api-reference/commands/runscript#running-conditionally). If this does not solve your use case, please file a feature request [here](https://github.com/mobile-dev-inc/maestro/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=%5BFeature+Request%5D). Thanks!
